File: TUnit.TestProject/DynamicCodeOnlyAttribute.cs
using System.Runtime.CompilerServices;
using TUnit.Core;
using TUnit.Core.Interfaces;

namespace TUnit.TestProject;

public class DynamicCodeOnlyAttribute() : SkipAttribute("This test is only supported when dynamic code is available")
{
    public override Task<bool> ShouldSkip(TestRegisteredContext context)
    {
#if NET
        return Task.FromResult(!RuntimeFeature.IsDynamicCodeSupported);
#else
        return Task.FromResult(false);
#endif
    }
}

File: TUnit.TestProject.FSharp/DynamicCodeOnlyAttribute.fs
namespace TUnit.TestProject.FSharp

open System.Threading.Tasks
open System.Runtime.CompilerServices
open TUnit.Core

// F# equivalent of DynamicCodeOnlyAttribute.cs

type DynamicCodeOnlyAttribute() =
    inherit SkipAttribute("This test is only supported when dynamic code is available")
    override _.ShouldSkip(context: BeforeTestContext) =
#if NET
        Task.FromResult(not RuntimeFeature.IsDynamicCodeSupported)
#else
        Task.FromResult(false)
#endif
